app/src/display/widgets/ble_passkey.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <lvgl.h>
+#include <stdio.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/ble.h>
+#include <zmk/ble/auth.h>
+#include <zmk/display.h>
+#include <zmk/display/widgets/ble_passkey.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/ble_auth_state_changed.h>
+
+static sys_slist_t widgets = SYS_SLIST_STATIC_INIT(&widgets);
+
+static void get_passkey_entry_text(char *str, size_t size, const struct zmk_ble_auth_state *state) {
+    if (state->cursor_index == 0) {
+        snprintf(str, size, "_");
+    } else if (state->cursor_index == 6) {
+        snprintf(str, size, "%06u " LV_SYMBOL_NEW_LINE, state->passkey);
+    } else {
+        snprintf(str, size, "%0*u_", state->cursor_index, state->passkey);
+    }
+}
+
+static void update_passkey_widget(struct zmk_widget_ble_passkey *widget,
+                                  const struct zmk_ble_auth_state *state) {
+    const int profile_index = state->profile_index + 1;
+    lv_label_set_text_fmt(widget->profile, LV_SYMBOL_WIFI " %i", profile_index);
+
+    switch (state->mode) {
+    case ZMK_BLE_AUTH_MODE_PASSKEY_CONFIRM:
+        lv_label_set_text_static(widget->title, "Confirm PIN");
+        lv_label_set_text_fmt(widget->passkey, "%06u " LV_SYMBOL_NEW_LINE, state->passkey);
+        break;
+
+    case ZMK_BLE_AUTH_MODE_PASSKEY_ENTRY: {
```

Comment:
This mode is typically for when only the host has a display and the keyboard does not, but since it is possible to configure things such that only passkey entry is enabled event though there is a display, I figured we might as well support this anyways.